### ISSUE TITLE
Favourite/star is not button

### ISSUE THREAD

**Issue by LukasHirt** [owncloud/web#1645]
## Description
Favourite/Star button is not a button, therefore cannot receive focus

**Comment by PVince81** [comment]
I've examined a few possible approaches:

## Approaches
### Approach 1

- update the "oc-star" element in the design system to include a new attribute "clickable". when set, wrap the icon with a "a href" element.
- in Phoenix, set the "clickable" attribute

### Approach 2

- same as approach 1 but add the "clickable" on the "oc-icon" element.

### Approach 3

- simply wrap the the "oc-star" inside of Phoenix with an "a" element.

### Approach 4

- change "oc-button" to also allow for removing the border

## Assumptions

- sometimes we want star icons that are not clickable

## Notes

- depending how far we want to go, we could also add a "pending" state to the star to provide visual feedback to fix https://github.com/owncloud/phoenix/issues/1903

I'm not sure yet how to decide what balance we want to have between having more logic in the design system or more in Phoenix.

@LukasHirt @felixheidecke

**Comment by PVince81** [comment]
in terms of reusability, one rule of thumb is to first add it in Phoenix (approach 3 + pending spinner) and then see later on how many times we'll want to reuse this logic.
one there is a clear requirement for reusability, refactor and move this into the design system

does that make sense ?

**Comment by LukasHirt** [comment]
I'd be actually in favour of approach 4 - button without border. We need to adjust button component anyway to have the opportunity to use it like this also in other places + having `a` tag for action that doesn't lead user to any different page shouldn't be.

**Comment by LukasHirt** [comment]
There is already link button in uikit - https://getuikit.com/docs/button#style-modifiers

**Comment by pascalwengerter** [comment]
Will get resolved by #5018

**Comment by pascalwengerter** [comment]
Fixed via #5000, coming to `master` soon
